packages/core/src/components/viewer-grid/viewer-grid.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  h,
+  Component,
+  Prop,
+  State,
+  Host,
+  Element,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import * as focusTrap from "focus-trap";
+
+type PanelSize = "small" | "medium" | "large";
+
+@Component({
+  tag: "dso-viewer-grid",
+  styleUrl: "viewer-grid.scss",
+  shadow: true,
+})
+export class ViewerGrid {
+  @Prop()
+  panelOpen = false;
+
+  @State()
+  panelSize: PanelSize = "large";
+
+  @Event()
+  closeAside!: EventEmitter<MouseEvent | KeyboardEvent>;
+
+  @Element()
+  host!: HTMLElement;
+
+  overlay!: HTMLDivElement;
+
+  overlaySlot!: HTMLDivElement;
+
+  trap!: focusTrap.FocusTrap;
+
+  shrinkPanel = () => {
+    this.panelSize = this.panelSize == "large" ? "medium" : "small";
+  };
+
+  expandPanel = () => {
+    this.panelSize = this.panelSize == "small" ? "medium" : "large";
+  };
+
+  updateFocusTrap() {
+    if (!this.trap) {
+      return;
+    }
+
+    if (this.panelOpen) {
+      this.trap.activate();
+      this.host.addEventListener("keydown", this.keyDownListener);
+    } else {
+      this.trap.deactivate();
+      this.host.removeEventListener("keydown", this.keyDownListener);
+    }
+  }
+
+  keyDownListener = (event: KeyboardEvent) => {
+    if (event.key != "Escape") {
+      return;
+    }
+
+    this.closeAside.emit(event);
+  };
+
+  connectedCallback() {
+    this.overlaySlot = this.host.querySelector(
+      "div[slot = 'overlay']"
+    ) as HTMLDivElement;
```

Comment:
De cast is onterecht: querySelector() kan ook `null` teruggeven.

Reply:
Done.